core/lib/spree/service_module.rb

```diff
@@ -23,6 +23,20 @@ def failure?
       end
     end
 
+    ResultError = Struct.new(:value) do
+      def to_s
+        if value&.respond_to?(:full_messages)
+          value.full_messages.join(', ')
+        elsif value.present?
+          value.to_s
+        end
+      end
+
+      def to_h
+        value.messages if value&.respond_to?(:messages)
```

Comment:
If `value&.respond_to?(:messages)` is falsey then `to_h` method will return `nil`. Maybe we could return `{}` in this case? Something like:

```
return value.messages if value&.respond_to?(:messages)

{}
```